batch/batches/ReachJobCleaner/KAsyncReachJobCleaner.php

```diff
@@ -0,0 +1,156 @@
+
+	/**
+	 * @param KSchedularTaskConfig $taskConfig
+	 */
+	public function __construct($taskConfig = null)
+	{
+		parent::__construct($taskConfig);
+		$this->reachClientPlugin = KalturaReachClientPlugin::get(self::$kClient);
+		$overdueTimePercentage = $this->getAdditionalParams('overdueTimePercentage');
+		$this->overdueTimePercentage = $overdueTimePercentage ? $overdueTimePercentage : 100;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::REACH_JOB_CLEANER;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::run()
+	*/
+	public function run($jobs = null)
+	{
+		KalturaLog::debug('Getting all the entry vendor task items');
+		$filter = new KalturaEntryVendorTaskFilter();
+		$filter->statusIn = KalturaEntryVendorTaskStatus::PENDING . ',' . KalturaEntryVendorTaskStatus::PROCESSING;
+		$filter->updatedAtLessThanOrEqual = time();
+		$filter->updatedAtGreaterThanOrEqual = time() -  (VendorServiceTurnAroundTime::TEN_DAYS + VendorServiceTurnAroundTime::FOUR_DAYS);
+		$filter->orderBy = KalturaEntryVendorTaskOrderBy::UPDATED_AT_ASC;
+
+		$pager = new KalturaFilterPager();
+		$pager->pageSize = self::MAX_PAGE_SIZE;
+		$pager->pageIndex = 1;
+
+		$lastUpdatedAt = 0;
+		$totalCount = 0;
+		do {
+			if ($lastUpdatedAt)
+			{
+				$filter->updatedAtGreaterThanOrEqual = $lastUpdatedAt;
+			}
+			try
+			{
+				$entryVendorTaskList = $this->reachClientPlugin->entryVendorTask->listAction($filter, $pager);
+				$tasksCount = count($entryVendorTaskList->objects);
+			}
+			catch (Exception $e)
+			{
+				KalturaLog::debug("Couldn't list entry Vendor Tasks " . $e->getMessage());
+				return;
+			}
+
+			$this->updateStuckJobs($entryVendorTaskList);
+
+			$totalCount += $tasksCount;
+			KalturaLog::debug("Handled More tasks - $tasksCount totalCount - " . $totalCount);
+			if ($tasksCount)
+			{
+				$lastObject = end($entryVendorTaskList->objects);
+				$lastUpdatedAt = $lastObject->updatedAt + 1;
+			}
+
+			unset($entryVendorTaskList);
+			if (function_exists('gc_collect_cycles')) // php 5.3 and above
+			{
+				gc_collect_cycles();
+			}
+		} while ($pager->pageSize == $tasksCount);
+
+		KalturaLog::debug('Done');
+	}
+
+	protected function updateStuckJobs($entryVendorTaskList)
+	{
+		foreach ($entryVendorTaskList->objects as $entryVendorTask)
+		{
+			if ($entryVendorTask->serviceType ==  KalturaVendorServiceType::HUMAN &&
+				$this->isDueDatePassed($entryVendorTask))
+			{
+				KalturaLog::debug('Going to update EntryVendorTask id ' . $entryVendorTask->id . ' to status ERROR. '
+					. self::JOB_DUE_DATE_EXCEEDED . $this->overdueTimePercentage . '%');
+
+				$kEntryVendorTask = new KalturaEntryVendorTask();
+				$kEntryVendorTask->status = KalturaEntryVendorTaskStatus::ERROR;
+				$kEntryVendorTask->errDescription = self::JOB_DUE_DATE_EXCEEDED . $this->overdueTimePercentage . '%';
+				try
+				{
+					KBatchBase::impersonate($entryVendorTask->partnerId);

```

Comment:
No need to do impersonate & impersonate before every call since there are no calls to the API in between.
You can just call impersonate once at the beginning of the function and one at the end